code/src/app/api/carInfo/route.ts

```diff
@@ -217,144 +166,105 @@ export async function GET(req: Request) {
 
-    if (existingLicense) {
+    if (!filteredData.carId) {
       return NextResponse.json(
-        { message: 'License number already registered.' },
+        { message: 'Please provide a carId' },
         { status: 400 },
       );
     }
+    await dbConnect();
 
-    // TODO Assumption: If new Date for last maintenance change, then also new mileage for last maintenance change
+    /* Check if license already exists for another car */
 
-    // Type conversion from request to Date object
-    let dateNextOilChangeDate = new Date('');
-    let dateNextTireChangeDate = new Date('');
+    if (filteredData.license) {
+      const existingCarWithSameLicense = await car.findOne({
+        license: filteredData.license,
+        _id: { $ne: filteredData.carId }, // Exclude the car being updated
+      });
 
-    // Oil Maintenance
-    if (
-      isMaintenanceTypeRequired(
-        mileageLastOilChange,
-        rotationMileageOil,
-        mileage,
-      )
-    ) {
-      dateNextOilChangeDate = new Date(dateNextOilChange);
-      oilOverdue = isMaintenanceOverdue(dateNextOilChangeDate, oilOverdue);
-      dateNextOilChangeDate = dateNeedsMaintenance(dateNextOilChangeDate);
-    } else {
-      oilOverdue = false;
-      dateNextOilChange = '';
+      if (existingCarWithSameLicense) {
+        return NextResponse.json(
+          { message: 'License already exists for another car.' },
+          { status: 401 },
+        );
+      }
     }
 
-    // Tire Maintenance
+    /* Check for new date and mileage for last maintenance change */
+
     if (
-      isMaintenanceTypeRequired(
-        mileageLastTireChange,
-        rotationMileageTire,
-        mileage,
-      )
+      (filteredData.dateNextOilChange && !filteredData.mileageLastOilChange) ||
+      (filteredData.dateNextTireChange && !filteredData.mileageLastTireChange)
     ) {
-      dateNextTireChangeDate = new Date(dateNextTireChange);
-      tireOverdue = isMaintenanceOverdue(dateNextTireChangeDate, tireOverdue);
-      dateNextTireChangeDate = dateNeedsMaintenance(dateNextTireChangeDate);
-    } else {
-      tireOverdue = false;
-      dateNextTireChange = '';
-    }
-
-    // set overwrites and type conversion to DB date string
-    maintenanceRequired = false;
-
-    if (dateNextOilChangeDate.getDate()) {
-      dateNextOilChange = dateNextOilChangeDate.toISOString();
-      maintenanceRequired = true;
-    }
-    if (dateNextTireChangeDate.getDate()) {
-      dateNextTireChange = dateNextTireChangeDate.toISOString();
-      maintenanceRequired = true;
+      return NextResponse.json(
+        {
+          message:
+            'If a new date for last maintenance change is provided, then a new mileage for the last maintenance change should also be provided.',
+        },
+        { status: 402 },
+      );
     }
+    const currentCarData = await car.findOne({ _id: filteredData.carId });
+    const updatedCarData = { ...currentCarData.toObject(), ...filteredData };
 
```

Comment:
Get previous mileage for DB. Becuase there is a situation, if user only changed mileage but not change maitenance info, the OilOverdue and TireOverDue would not update. So get previous mileage from db, and compare with current to check if overdue or not